src/1_utils.js

```diff
@@ -733,3 +733,43 @@ utils.getAdditionalMetadata = function() {
 	metadata = utils.addPropertyIfNotNull(metadata, "canonical_url", utils.getCanonicalURL());
 	return metadata && Object.keys(metadata).length > 0 ? metadata : null;
 };
+
+// v2/event utility functions
+
+var BRANCH_STANDARD_EVENTS = [ 'ADD_TO_CART', 'ADD_TO_WISHLIST', 'VIEW_CART', 'INITIATE_PURCHASE', 'ADD_PAYMENT_INFO', 'PURCHASE', 'SPEND_CREDITS', 'SEARCH', 'VIEW_CONTENT', 'VIEW_CONTENT_LIST', 'RATE', 'SHARE_CONTENT', 'COMPLETE_REGISTRATION', 'COMPLETE_TUTORIAL', 'ACHIEVE_LEVEL', 'UNLOCK_ACHIEVEMENTS' ];
+var BRANCH_STANDARD_EVENT_DATA = [ 'transaction_id', 'revenue', 'currency', 'shipping', 'tax', 'coupon', 'affiliation', 'search_query', 'description', 'product_categories' ];
+
+var isStandardEvent = function(eventName) {
+	return BRANCH_STANDARD_EVENTS.indexOf(eventName) > -1;
+};
+
+utils.determineV2EventType = function(eventName) {
+	return isStandardEvent(eventName) ? 'STANDARD' : 'CUSTOM';
+};
+
+utils.separateEventAndCustomData = function(eventAndCustomData) {
+
+	if (!eventAndCustomData || Object.keys(eventAndCustomData).length === 0) {
+		return null;
+	}
+	var customDataKeys = utils.calculateDiffBetweenArrays(BRANCH_STANDARD_EVENT_DATA, Object.keys(eventAndCustomData));
+	var customData = {};
+
+	for (var i = 0; i < customDataKeys.length; i++) {
+		var key = customDataKeys[i];
+		customData[key] = eventAndCustomData[key];
+		delete eventAndCustomData[key];
+	}
+
+	return {
+		custom_data: customData,
+		event_data: eventAndCustomData
+	};
+};
+
+utils.validateParameterType = function(parameter, type) {
```

Comment:
I would first check that the parameter exists incase you are looking to validate an object and null get's passed in. `typeof null === 'object'` turns out to be true